src/network/tc/etf.c

```diff
@@ -0,0 +1,262 @@
+        EarliestTxTimeFirst *etf;
+        __s32 v;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = qdisc_new_static(QDISC_KIND_ETF, network, filename, section_line, &qdisc);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                                  "More than one kind of queueing discipline, ignoring assignment: %m");
+
+        etf = ETF(qdisc);
+
+        if (isempty(rvalue)) {
+                etf->delta = 0;
+
+                TAKE_PTR(qdisc);
+                return 0;
+        }
+
+        r = safe_atoi32(rvalue, &v);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse '%s=', ignoring assignment: %s",
+                           lvalue, rvalue);
+                return 0;
+        }
+        if (v > INT32_MAX) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Specified '%s=' is too large, ignoring assignment: %s",
+                           lvalue, rvalue);
+                return 0;
+        }
+
+        etf->delta = v;
+
+        TAKE_PTR(qdisc);
+
+        return 0;
+}
+
+
+int config_parse_tc_etf_bool (
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(qdisc_free_or_set_invalidp) QDisc *qdisc = NULL;
+        Network *network = data;
+        EarliestTxTimeFirst *etf;
+        int r;
+        bool k, *p;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = qdisc_new_static(QDISC_KIND_ETF, network, filename, section_line, &qdisc);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                                  "More than one kind of queueing discipline, ignoring assignment: %m");
+
+        etf = ETF(qdisc);
+
+        if (streq(lvalue, "EarliestTxTimeFirstDeadline"))
+                p = &etf->deadline;
+        else if (streq(lvalue, "EarliestTxTimeFirstOffload"))
+                p = &etf->offload;
+        else if (streq(lvalue, "EarliestTxTimeFirstSkipsock"))
+                p = &etf->skipsock;
+        else
+                assert_not_reached();
+
+        if (isempty(rvalue)) {
+
+                *p = -1;
+
+                TAKE_PTR(qdisc);
+                return 0;
+        }
+
+        k = parse_boolean(rvalue);
+        if (k < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
```

Comment:
you pass `r` as error here, but I guess you mean `k`...

btw, why bother with `k` at all? why not use `r` here for everything?